controllers/attester_controller.go

```diff
@@ -56,77 +59,204 @@ func (r *AttesterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+        secret := &corev1.Secret{}
+        err = r.Get(ctx, req.NamespacedName, secret)
+        if err != nil {
+            log.Error(err, "Failed to get the secret")
+        }
+
+        err = r.Delete(ctx, secret)
+        if err != nil {
+            log.Error(err, "Failed to delete the secret")
+        }
+
+        // Deleting attester object
+        delete(r.Attesters, req.Name)
+
+        return ctrl.Result{}, err
+    }
+
+    // If there are 0 conditions then initialize conditions by adding two with unknown statuses
+    if len(att.Status.Conditions) == 0 {
+
+        policyCondition := rodev1.AttesterCondition{
+            Type: rodev1.AttesterConditionCompiled,
+            Status: rodev1.ConditionUnknown,
+        }
+
+        att.Status.Conditions = append(att.Status.Conditions, policyCondition)
+
+        secretCondition := rodev1.AttesterCondition{
+            Type: rodev1.AttesterConditionSecret,
+            Status: rodev1.ConditionUnknown,
+        }
+
+        att.Status.Conditions = append(att.Status.Conditions, secretCondition)
+
+        if err := r.Status().Update(ctx, att); err != nil {
+            log.Error(err, "Unable to initialize attester status")
+            return ctrl.Result{}, err
+        }
+    }
 
-	// TODO: check if secret already exists before doing this...maybe just load secret
+    // Always recompile the policy
+    policy, err := attester.NewPolicy(req.Name, att.Spec.Policy, opaTrace)
+    if err != nil {
+        log.Error(err, "Unable to create policy")
+        att.Status.Conditions[0].Status = rodev1.ConditionFalse
 
-	signerSecret := &corev1.Secret{}
+        if err := r.Status().Update(ctx, att); err != nil {
+            log.Error(err, "Unable to update Attester status")
+            return ctrl.Result{}, err
+        }
+        return ctrl.Result{}, err
+    }
+
+    att.Status.Conditions[0].Status = rodev1.ConditionTrue
+
+    if err := r.Status().Update(ctx, att); err != nil {
+        log.Error(err, "Unable to update Attester status")
+        return ctrl.Result{}, err
+    }
+
+    signerSecret := &corev1.Secret{}
 	var signer attester.Signer
 
-    if att.Spec.PgpSecret == req.Name {
-        err := r.Get(ctx, client.ObjectKey{
-            Namespace: req.Namespace,
-            Name: att.Spec.PgpSecret,
-            }, signerSecret)
+    // Check that the secret exists, if it does, recreate a signer from the secret
+    if att.Status.Conditions[1].Status != "True" {
+
+        err = r.Get(ctx, req.NamespacedName, signerSecret)
         if err != nil {
-		    log.Error(err, "Unable to load signerSecret")
-		    return ctrl.Result{}, err
-        }
-	} else {
-        signer, err = attester.NewSigner(req.Name)
-		if err != nil {
-			log.Error(err, "Unable to create signer")
-			return ctrl.Result{}, err
-		}
-        log.Info("Created the signer successfully")
 
-        publicKey := signer.KeyID()
+            // If the secret wasn't found then create the secret
```

Comment:
I think the logic for managing an attester's secrets should be pulled out into another package, perhaps in a new file within `pkg/attester`.